### PR TITLE
Hello World! - Simplifying sample for new users and adding snippet wrappers

### PR DESCRIPTION
**Pull request by cullywakelin**
This PR aims to simplify the "helloworld" Go SDK sample to use in conjunction with a "helloworld" Go SDK tutorial.

Snipsync wrappers have been added to the relevant functions.
Comments have been added to ensure each step in the code can be easily understood.
Unrelated files have been removed from the sample directory.
Files have been renamed for comprehension.

### CODE REVIEW

File: helloworld/README.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-2) Run the following command to start the worker
+## HelloWorld! Temporal Go SDK sample
+
+To use this sample, first [Install and run the Temporal server](https://docs.temporal.io/docs/install-temporal-server).
+
+Run the Workflow:
+
 ```
-go run helloworld/worker/main.go
+go run helloworld/starter/main.go
```
Comment (sergeybykov) [inline helloworld/README.md:8]: Why flip the order? To show that starter will be blocked until the worker starts?

File: helloworld/starter/main.go [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
 import (
-	"context"
-	"log"
-
-	"go.temporal.io/sdk/client"
-
-	"github.com/temporalio/temporal-go-samples/helloworld"
+  "context"
+  "log"
+  // Import the Temporal Go SDK client package
```
Comment (mfateev) [inline helloworld/starter/main.go:7]: Aren't such comments just noise?

File: helloworld/worker/main.go [hunk 3]
```diff
@@ -1,29 +1,35 @@
...
+  // Task Queue that the Worker will listen to
+  // Must be the same name as the Task Queue the Workflow is sent to
+  taskQueue := "helloworld-task-queue"
+  // Create a Worker that is listening to the taskQueue
+  wrkr := worker.New(c, taskQueue, worker.Options{})
+  // Register Workflow with the Worker
+  wrkr.RegisterWorkflow(helloworld.HelloWorldWorkflow)
+  // Register Activity with the Worker
+  wrkr.RegisterActivity(helloworld.HelloWorldActivity)
+  // Run the Worker
```
Comment (mfateev) [inline helloworld/worker/main.go:30]: until the interrupt signal is received

File: helloworld/README.md (hunk 1 above)
Comment (cullywakelin) [reply helloworld/README.md:8]: Yes.

File: helloworld/starter/main.go (hunk 2 above)
Comment (cullywakelin) [reply helloworld/starter/main.go:7]: Ya, probably. Will remove these.

**Comment by cullywakelin** [comment]
Using template repo instead

**Pull request closed without merging**